### PR TITLE
Code coverage fixes

### PR DESCRIPTION
**Pull request by paulkagiri**
This PR aims to address 3 main issues:
- To fix the exception being raised when one tries to run instrumented tests with code coverage turned on.
    ```
    java.lang.NoClassDefFoundError: Failed resolution of: Lorg/jacoco/agent/rt/internal_3570298/Offline;
    at com.microsoft.identity.client.Logger.$jacocoInit(Unknown Source:13)
    at com.microsoft.identity.client.Logger.<clinit>(Unknown Source:0)
    ...
    ```

- Adding the ability to turn code coverage on or off easily without affecting the rest of the pipelines - https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1635/files#diff-824c1e8ad11e20caed0bec7162a99779b9a4bcf1178d99fae3e39f69889f8959R21 & https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1635/files#diff-de2d54ff1759d86cbb48be44d1dc5757fbc00c1775efea9862854297630c9676R31

- Moving the code coverage pipeline from Github Actions to Azure Pipelines - this is because there was the need to run e2e tests which require the test automation credentials only available to Azure Pipelines

### CODE REVIEW

**Comment by AdamBJohnsonx** [comment]
Should we see a change to use the build property here?

**Comment by paulkagiri** [comment]
> Should we see a change to use the build property here?

This has been added

File: .github/workflows/codecov.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on:
   pull_request:
 
 env:
-  ROBOELECTRIC_SDK_VERSION: 29
+  ROBOLECTRIC_SDK_VERSION: 28
```
Comment (shahzaibj) [inline .github/workflows/codecov.yml:7]: Why changing this?

File: azure-pipelines/code-coverage/common.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ jobs:
   - task: Gradle@2
     displayName: Run common code coverage
     inputs:
-      tasks: common:localDebugCommonUnitTestCoverageReport -Plabtest -ProbolectricSdkVersion=${{variables.robolectricSdkVersion}}
```
Comment (shahzaibj) [inline azure-pipelines/code-coverage/common.yml:41]: why remove `-Plabtest`?

Comment (shahzaibj) [reply azure-pipelines/code-coverage/common.yml:41]: we should know about lab tests in coverage reports

Comment (paulkagiri) [reply azure-pipelines/code-coverage/common.yml:41]: See https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1554#discussion_r758464826

**Comment by rpdome** [comment]
Could you please give a brief description of the issue (root cause), and how you fixed it?

**Comment by paulkagiri** [comment]
> Could you please give a brief description of the issue (root cause), and how you fixed it?

This is already done, kindly read through the PR description and if it still doesn't make sense I will add more elaborations

**Review by rpdome — APPROVED** [review]
LGTM. Thanks Paul.

File: azure-pipelines/code-coverage/common.yml [hunk 3]
```diff
@@ -1,12 +1,17 @@
...
 # Variable: 'commonBranchName' common branch to be used when running the pipeline manually
 name: $(date:yyyyMMdd)$(rev:.r)
 
 trigger: none
 
+pr:
+  branches:
+    include:
+      - '*'
+
```
Comment (p3dr0rv) [inline azure-pipelines/code-coverage/common.yml:14]: f no pr triggers appear in your YAML file, pull request validations are automatically enabled for all branches, as if you wrote the following pr trigger. This configuration triggers a build when any pull request is created, and when commits come into the source branch of any active pull request.

Comment (paulkagiri) [reply azure-pipelines/code-coverage/common.yml:14]: This is the desired behaviour so that's ok.

**Review by p3dr0rv — APPROVED** [review]

**Review by shoatman — APPROVED** [review]

**Pull request merged**